.github/workflows/main.yml

```diff
@@ -0,0 +1,272 @@
+
+      - name: install/configure dependencies, ch-grow
+        if: ${{ matrix.builder == 'ch-grow' }}
+        run: |
+          # Use most current Lark rather than the one in Ubuntu b/c new
+          # versions sometimes break things.
+          sudo pip3 install lark-parser
+
+      - name: install/configure dependencies, all Buildah
+        if: ${{ startsWith(matrix.builder, 'buildah') }}
+        run: |
+          command -v buildah
+          buildah --version
+          command -v runc
+          runc --version
+          # As of 2020-11-30, stock registries.conf is pretty simple; it
+          # includes Docker Hub (docker.io) and then quay.io. Still, use ours
+          # for stability.
+          cat /etc/containers/registries.conf
+          cat <<'EOF' | sudo tee /etc/containers/registries.conf
+          [registries.search]
+            registries = ['docker.io']
+          EOF
+
+      - name: install/configure dependencies, privileged Buildah
+        if: ${{ startsWith(matrix.builder, 'buildah-') }}
+        run: |
+          sudo usermod --add-subuids 10000-65536 $USER
+          sudo usermod --add-subgids 10000-65536 $USER
+
+      - name: build/install from Git
+        run: |
+          ./autogen.sh
+          # Remove Autotools to make sure everything works without them.
+          sudo apt-get remove autoconf automake
+          # Configure and verify output.
+          ./configure --prefix=$ch_prefix/from-git
+          set -x
+          fgrep 'documentation: yes' config.log
+          [[ $CH_BUILDER = buildah* ]] && fgrep 'with Buildah: yes' config.log
+          [[ $CH_BUILDER = docker ]] && fgrep 'with Docker: yes' config.log
+          [[ $CH_BUILDER = ch-grow ]] && fgrep 'with ch-grow(1): yes' config.log
+          fgrep 'at least one builder ... yes' config.log
+          fgrep 'ch-run(1): yes' config.log
+          fgrep 'complete test suite: yes' config.log
+          set +x
+          # Build and install.
+          make $ch_makej
+          sudo make $ch_makej install
+          bin/ch-run --version
+          $ch_prefix/from-git/bin/ch-run --version
+
+      - name: late setup & validation
+        run: |
+          bin/ch-test --is-pedantic all
+          bin/ch-test --is-sudo all
+
+      - name: make filesystem permissions fixtures
+        run: |
+          bin/ch-test mk-perm-dirs
+
+      - name: configure sudo to user root, group non-root
+        if: ${{ matrix.keep_sudo }}
+        run: |
+          sudo sed -Ei 's/=\(ALL\)/=(ALL:ALL)/g' /etc/sudoers.d/runner
+          sudo cat /etc/sudoers.d/runner
+
+      - name: remove sudo
+        if: ${{ ! matrix.keep_sudo }}
+        run: |
+          sudo rm /etc/sudoers.d/runner
+          ! sudo echo hello
+
+      - name: build/install from tarball
+        if: ${{ matrix.builder == 'docker' && matrix.keep_sudo }}
+        run: |
+          # Create and unpack tarball.
+          make $ch_makej dist
+          mv charliecloud-*.tar.gz $ch_prefix
```

Comment:
I don't think so. I think the reasoning for the wildcard is that it saves the trouble of getting the version into a variable. If there is more than one tarball in there, `tar` and/or `cd` will break. And since we start with a clean VM, I'm not sure how we would get the wrong version. I will add a comment about that assumption.